src/userspace/exec_monitor.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Google LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+#include <bpf/libbpf.h>
+#include "exec_monitor.skel.h"
+#include <unistd.h>
+#include <bpf/bpf.h>
+#include "exec_monitor.h"
+
+static int process_sample(void *ctx, void *data, size_t len)
+{
+	struct process_info *s = data;
```

Comment:
When dereferencing a pointer in C you should always be very careful with the size of the output. It might be that in 99.99% of the cases the buffer you get in data contains a complete struct process_info and your code works but maybe there is a bug somewhere, the buffer is trimmed or an extra message that you never anticipated happens on the ringbuffer and you'll try to access fields out of the buffer. It's a good practice in C to check the size of the buffers you get before you access them. Something like if(len < sizeof(struct process_info)) return error; this will make your program more resilient. Maybe this particular case doesn't really need it but this is still a best practice that you should follow in C.